### PR TITLE
feat: add opt-in LLM branch resume briefs

### LINKED ISSUE #32: Research: richer session summaries via the user's existing Claude Code (claude -p / Agent SDK), async + deterministic fallback
## Description

The per-branch `context_summary` is built by deterministic Python (`summarizer.py`, no LLM): topic = first user message (truncated), first-2 + last-6 exchanges (each mid-truncated), metadata, and the omitted middle represented only by file basenames. It's a mechanical *extract of the endpoints*, not an abstract of what the session was about. That limits the quality of (a) the SessionStart context injection and (b) the entrypoint-A "find sessions like" result cards — especially for long sessions whose substance lives in the middle.

**Research** generating a richer summary using the **user's existing Claude Code install** — `claude -p "<summarize this transcript>"` or the Claude Agent SDK — rather than a separate API integration. Produce a recommendation + small prototype, not a committed implementation.

This is a **research/spike** issue. It is complementary to (not a substitute for) chunk-level embedding: chunking fixes *search recall*; a better summary improves *display/injection quality*. If chunking lands, the summary is no longer a search target, so this is purely about the human/agent-facing card and the injected context.

## Why the obvious objections don't apply

The host environment is Claude Code, which already provides the LLM, auth, and network:
- **No API key / billing setup** — `claude -p` / the SDK use the user's existing Claude Code credentials.
- **Offline isn't a concern** — the transcript exists because Claude Code was online; a just-ended session was necessarily networked.
- **No added privacy exposure** — the conversation content already transited Anthropic to be generated.

What genuinely remains to investigate: **latency** (spawning `claude -p` costs seconds, so it must not sit on the inline Stop-hook path — see `sync_current.py` synchronous `embed_branch`), **non-determinism** (need a fallback), and **usage quota** (it spends the user's own Claude usage).

## Acceptance Criteria (research deliverable)

- [ ] A written recommendation: adopt / adopt-with-conditions / don't, with rationale.
- [ ] **Latency measured**: time a `claude -p` summarization of a representative transcript (and an Agent SDK call), to decide placement.
- [ ] **Placement decided**: confirmed off the inline Stop-hook hot path — async backfill (cf. `backfill_summaries.py`) and/or detached spawn — with the deterministic summary as the instant default/fallback.
- [ ] **Invocation method chosen**: `claude -p` vs Agent SDK, with the tradeoff (simplicity vs structured output / control).
- [ ] **Output shape defined**: does the LLM summary populate the existing `context_summary` / `context_summary_json` shape (so `render_context_summary` and consumers are unchanged), or a new field? Must capture the *middle*, not just endpoints.
- [ ] **Cost/opt-in policy**: opt-in vs default-on; whether to only summarize sessions above N exchanges (short ones are fine deterministically); how the knob is configured (`config.json` / `write_config`).
- [ ] **Fallback proven**: when `claude -p`/SDK is unavailable or fails, the deterministic summary is used and nothing breaks.
- [ ] A minimal prototype prompt + one before/after example on a real long session.

## Affected Areas (investigation scope, not commitment)

- `src/ccrecall/summarizer.py` — current deterministic summary; where an LLM upgrade would slot in or be called from.
- `src/ccrecall/hooks/backfill_summaries.py` — the existing async/opt-in summary path; natural home for an LLM pass.
- `src/ccrecall/hooks/sync_current.py` — confirm the LLM call stays *off* this inline path.
- `src/ccrecall/hooks/write_config.py` / `config.json` — the opt-in/cost knob.
- `src/ccrecall/db.py` / `schema.py` — `summary_version` gating if the summary shape changes (re-summarize trigger).

## Context

Today's mechanical summary: `FIRST_EXCHANGES=2`, `LAST_EXCHANGES=6`, `truncate_mid` front-300/back-600; `_build_gap_summary` reduces the middle to file basenames. Stop-hook latency is actively protected (hooks are separate console scripts to keep the path ~440ms vs ~1800ms, per CLAUDE.md), which is why an LLM call must be async. Relationship: pairs with the chunk-level embedding issue — that one owns search recall, this one owns display/injection quality.

### PR DESCRIPTION
**Pull request by NodeJSmith**
### Branch Resume Briefs
- Add opt-in `ccrecall backfill llm-summaries` support so users can generate richer Branch Resume Briefs through their existing Claude Code auth without changing the deterministic summary path for everyone else.
- Surface valid cached enrichments above the deterministic session summary in startup context and hydrated search/recent views so the extra detail helps continuation, but stale or invalid enrichments quietly fall back to the existing baseline.

### Safety, Gating, and Source Provenance
- Gate every Claude-powered path behind an explicit capability check and the existing `llm_summaries_enabled` opt-in so ccrecall does not create self-importing transcripts or unexpectedly send transcript packets upstream.
- Track transcript-source provenance, source hashes, enrichment status, and update timestamps in the DB so cached briefs are only reused when the deterministic summary and transcript inputs still match the active branch.
- Harden transcript discovery around unsafe paths, missing files, and detached worker failure modes so enrichment exits conservatively instead of weakening the core import/sync flow.

### CLI, Docs, and Evaluation
- Add the manual backfill command, selectors, status plumbing, and detached sync-worker spawn path so enrichment can be run intentionally for history or automatically after a successful Stop-hook sync.
- Document the opt-in model, capability check, budget semantics, and internal worker behavior in `README.md`; see the `design/specs/008-llm-summary-enrichment/` docs for the design and evaluation details.
- Extend the test suite with end-to-end coverage for the contract, worker orchestration, rendering, CLI behavior, search hydration, and transcript-source edge cases so the new path can evolve without regressing the deterministic baseline.

Closes #32

## Summary by CodeRabbit

* **New Features**
  * Added opt-in Claude-powered enrichment for Branch Resume Briefs.
  * Added automatic background enrichment and manual backfill with filtering, retry, budget, and capability-check options.
  * Enriched search cards and context can display titles and “where we left off” previews.

* **Bug Fixes**
  * Improved nested transcript discovery while rejecting unsafe symlink paths.
  * Added source tracking, validation, stale-result protection, and deterministic fallback behavior.

* **Documentation**
  * Documented configuration, privacy, budgets, workflows, evaluation procedures, and fallback behavior.

### CODE REVIEW

File: src/ccrecall/llm_summarizer.py [hunk 1]
```diff
@@ -0,0 +1,815 @@
...
+        return None, _cap(completed.stderr.strip() or completed.stdout.strip() or "claude --version failed")
+    output = (completed.stdout or completed.stderr).strip()
+    return output.splitlines()[0] if output else "unknown", None
+
+
+def _parse_claude_stdout(stdout: str) -> dict[str, Any]:
+    data = json.loads(stdout)
+    if not isinstance(data, dict):
+        raise ValueError("stdout must be a json object")
+    return data
```
[the remark this answers is not in the document]
Comment (NodeJSmith) [reply src/ccrecall/llm_summarizer.py:309]: Fixed - successful Claude CLI result envelopes now unwrap structured_output before schema validation.

File: src/ccrecall/hooks/backfill_llm_summaries.py [hunk 2]
```diff
@@ -0,0 +1,713 @@
...
+            "source_transcript_paths": [str(path) for path in source_files],
+        }
+        project_root = (
+            Path(row["project_root"]) if isinstance(row["project_root"], str) and row["project_root"] else None
+        )
+        valid_file_paths = _parse_json_string_list(row["files_modified"]) | branch_content_file_paths(
+            source_files,
+            active_branch_uuids,
+            project_root=project_root,
+        )
```
[the remark this answers is not in the document]
Comment (NodeJSmith) [reply src/ccrecall/hooks/backfill_llm_summaries.py]: Fixed - transcript file paths under the project root are normalized before response allowlist validation.

File: design/specs/008-llm-summary-enrichment/evaluation.md [hunk 3]
```diff
@@ -0,0 +1,215 @@
...
+                   b.summary_enrichment_version
+            FROM branches b
+            JOIN sessions s ON s.id = b.session_id
+            JOIN projects p ON p.id = s.project_id
+            WHERE p.name = ? AND b.is_active = 1
+            """,
+            (scenario["id"],),
+        ).fetchone()
+        if row is None:
+            raise SystemExit(f"missing imported branch for {scenario['id']}")
```
[the remark this answers is not in the document]
Comment (NodeJSmith) [reply design/specs/008-llm-summary-enrichment/evaluation.md]: Fixed - the evaluation recipe looks up projects by their materialized fixture stems.

File: src/ccrecall/db.py [hunk 4]
```diff
@@ -11,15 +11,14 @@
 
 import sqlite_vec
 
-from ccrecall.config import DEFAULT_DB_PATH, ensure_parent_dir, get_db_path
+import ccrecall.llm_summary_db as llm_summary_db
```
[the remark this answers is not in the document]
Comment (NodeJSmith) [reply src/ccrecall/db.py]: Fixed - changed the redundant llm_summary_db alias import to the package import form.

File: src/ccrecall/embed_ops.py [hunk 5]
```diff
@@ -40,6 +42,11 @@ def write_branch_summary(cursor: sqlite3.Cursor, branch_db_id: int) -> str | Non
             """,
             (summary_md, summary_json, SUMMARY_VERSION, branch_db_id),
         )
+        summary_source_hash = compute_branch_summary_source_hash(cursor, branch_db_id)
+        cursor.execute(
+            "UPDATE branches SET summary_source_hash = ? WHERE id = ?",
+            (summary_source_hash, branch_db_id),
+        )
```
[the remark this answers is not in the document]
Comment (NodeJSmith) [reply src/ccrecall/embed_ops.py]: Fixed - summary and source-hash persistence now share a savepoint, preserving backfill eligibility on failure.

File: src/ccrecall/hooks/backfill_llm_summaries.py [hunk 6]
```diff
@@ -0,0 +1,713 @@
...
+    if _is_capability_gated_budget_failure(row):
+        return capability_status != status
+    if status in FORCE_ONLY_STATUSES:
+        return False
+    if status == STATUS_UNSAFE_SOURCE_PATH:
+        return False
+    if status in CAPABILITY_BLOCKED_STATUSES:
+        return capability_status != status
+    if status in SOURCE_STATUSES:
+        return source_status == STATUS_OK
```
[the remark this answers is not in the document]
Comment (NodeJSmith) [reply src/ccrecall/hooks/backfill_llm_summaries.py:325]: Fixed - removed the unreachable unsafe-source status membership.

File: src/ccrecall/hooks/backfill_llm_summaries.py [hunk 7]
```diff
@@ -0,0 +1,713 @@
...
+                    total_success += 1
+                if result.selected:
+                    selected_branches += 1
+                if limit is not None and selected_branches >= limit:
+                    return complete()
+            if current_session:
+                break
+    except (sqlite3.Error, OSError):
+        logger.exception("LLM summary worker aborted")
+        return EXIT_ABORT
```
[the remark this answers is not in the document]
Comment (NodeJSmith) [reply src/ccrecall/hooks/backfill_llm_summaries.py:679]: Fixed - unexpected detached-worker exceptions are logged and return the abort status.

File: src/ccrecall/hooks/backfill_llm_summaries.py [hunk 8]
```diff
@@ -0,0 +1,713 @@
...
+    parser = argparse.ArgumentParser(prog="ccrecall-llm-summaries")
+    parser.add_argument("--days", type=int)
+    parser.add_argument("--limit", type=int)
+    parser.add_argument("--session")
+    parser.add_argument("--current-session", action="store_true")
+    parser.add_argument("--force", action="store_true")
+    parser.add_argument("--verbose", action="store_true")
+    args = parser.parse_args(argv)
+    if args.current_session and args.session is None:
+        parser.error("--current-session requires --session")
```
[the remark this answers is not in the document]
Comment (NodeJSmith) [reply src/ccrecall/hooks/backfill_llm_summaries.py:729]: Fixed - invalid direct worker limits now produce argparse errors.

File: src/ccrecall/hooks/subprocess_utils.py [hunk 9]
```diff
@@ -0,0 +1,15 @@
...
+
+
+def detached_popen_kwargs() -> dict[str, Any]:
+    """Return the cross-platform kwargs used for detached hook subprocesses."""
+    kwargs: dict[str, Any] = {"stdout": subprocess.DEVNULL, "stderr": subprocess.DEVNULL}
+    if sys.platform == "win32":
+        kwargs["creationflags"] = subprocess.CREATE_NEW_PROCESS_GROUP | subprocess.DETACHED_PROCESS
+    else:
+        kwargs["start_new_session"] = True
+    return kwargs
```
[the remark this answers is not in the document]
Comment (NodeJSmith) [reply src/ccrecall/hooks/subprocess_utils.py]: Fixed - detached subprocesses now redirect stdin to DEVNULL.

File: src/ccrecall/hooks/sync_current.py [hunk 10]
```diff
@@ -87,39 +91,39 @@ def validate_session_id(session_id: str) -> bool:
...
 
 
-def _is_under(path: Path, base: Path) -> bool:
-    """Check whether path resolves to a location under base (symlink-escape guard)."""
+def get_session_file(projects_dir: Path, session_id: str) -> Path | None:
+    """Find the JSONL file for a session ID. Validates path stays under projects_dir."""
+    discovery = discover_session_transcript_files(projects_dir, session_id)
+    if discovery.files:
+        return discovery.files[0]
+    return None
```
[the remark this answers is not in the document]
Comment (NodeJSmith) [reply src/ccrecall/hooks/sync_current.py:101]: Fixed - sync rejects matching unsafe transcript sources while allowing safe sessions despite unrelated unsafe projects.

File: src/ccrecall/llm_summarizer.py [hunk 11]
```diff
@@ -0,0 +1,815 @@
...
+
+def _canonical_json(value: object) -> str:
+    return json.dumps(value, ensure_ascii=False, separators=(",", ":"), sort_keys=True)
+
+
+def capability_fingerprint(packet_dir: Path | None = None) -> str:
+    packet_dir = packet_dir or Path("<packet-dir>")
+    security_shape = _build_capability_check_argv(packet_dir)[1:-1]
+    normalized_shape = ["<packet-dir>" if arg == str(packet_dir) else arg for arg in security_shape]
+    return hashlib.sha256("\0".join(normalized_shape).encode("utf-8")).hexdigest()
```
[the remark this answers is not in the document]
Comment (NodeJSmith) [reply src/ccrecall/llm_summarizer.py]: Fixed - the capability fingerprint now derives from the shared capability-check security and persistence argv.

File: src/ccrecall/llm_summarizer.py [hunk 12]
```diff
@@ -0,0 +1,815 @@
...
+    if "rate limit" in text:
+        return STATUS_RATE_LIMITED, _cap(stderr or stdout)
+    if "budget" in text:
+        return STATUS_BUDGET_EXCEEDED, _cap(stderr or stdout)
+    if stdout:
+        try:
+            json.loads(stdout)
+        except ValueError:
+            return STATUS_INVALID_OUTPUT, _cap("invalid json on stdout")
+    return STATUS_ERROR, _cap(stderr or stdout or "claude failed")
```
[the remark this answers is not in the document]
Comment (NodeJSmith) [reply src/ccrecall/llm_summarizer.py:286]: Fixed - rate-limit and budget diagnostics take precedence and auth matching is bounded.

File: src/ccrecall/llm_summarizer.py [hunk 13]
```diff
@@ -0,0 +1,815 @@
...
+                text = result.get("text")
+                if isinstance(text, str):
+                    paths.update(_path_references(text, project_root=project_root))
+            for invocation in _tool_invocations(entry.get("message", {}).get("content")):
+                for signal in invocation.get("file_signals", []):
+                    if isinstance(signal, str):
+                        normalized_signal = _normalized_path_reference(signal, project_root=project_root)
+                        if normalized_signal is not None:
+                            paths.add(normalized_signal)
+    return paths
```
[the remark this answers is not in the document]
Comment (NodeJSmith) [reply src/ccrecall/llm_summarizer.py:646]: Not changing this path: retaining focused readers avoids holding normalized sensitive transcript data longer without measured benefit.

File: src/ccrecall/llm_summary_db.py [hunk 14]
```diff
@@ -0,0 +1,324 @@
...
+    migrate_to_v2: Callable[[sqlite3.Connection], None] = _migrate_to_v2,
+) -> None:
+    """Apply version-gated schema migrations up to SCHEMA_VERSION, atomically."""
+    current = conn.execute("PRAGMA user_version").fetchone()[0]
+
+    _migrate_to_v3(conn)
+    _migrate_to_v4(conn)
+    _migrate_to_v5(conn)
+    _migrate_to_v6(conn)
+    _migrate_to_v7(conn)
```
[the remark this answers is not in the document]
Comment (NodeJSmith) [reply src/ccrecall/llm_summary_db.py:272]: Fixed - v7 migration now skips ALTER attempts when all enrichment columns already exist and handles partial schemas.

File: src/ccrecall/parsing.py [hunk 15]
```diff
@@ -85,6 +85,22 @@ def parse_lines_with_uuids(lines: Iterable[str]) -> Generator[dict, None, None]:
...
 
 
+def parse_lines_with_uuids_and_numbers(lines: Iterable[str]) -> Generator[tuple[int, dict], None, None]:
+    """Yield line-numbered parsed JSONL entries that carry a uuid."""
+    for line_number, line in enumerate(lines, start=1):
+        line = line.strip()
+        if not line:
+            continue
+        try:
+            entry = json.loads(line)
```
[the remark this answers is not in the document]
Comment (NodeJSmith) [reply src/ccrecall/parsing.py]: Fixed - parsing now keeps the loop variable and uses a separate stripped value.

File: src/ccrecall/summary_enrichment.py [hunk 16]
```diff
@@ -0,0 +1,735 @@
...
+        for item in section.get("source_uuids", [])
+        if isinstance(item, str)
+    }
+    valid_file_paths = {
+        path
+        for section in body.get("files_and_reasons", [])
+        if isinstance(section, dict)
+        for path in [section.get("path")]
+        if isinstance(path, str)
+    }
```
[the remark this answers is not in the document]
Comment (NodeJSmith) [reply src/ccrecall/summary_enrichment.py:537]: Fixed - malformed stored enrichment shapes now safely fall back to deterministic rendering.

File: src/ccrecall/summary_enrichment.py [hunk 17]
```diff
@@ -0,0 +1,735 @@
...
+        """
+        SELECT b.leaf_uuid, b.summary_version, b.context_summary_json, b.aggregated_content,
+               b.exchange_count, b.started_at, b.ended_at, b.files_modified,
+               b.tool_counts, b.commits, s.git_branch
+        FROM branches b
+        JOIN sessions s ON b.session_id = s.id
+        WHERE b.id = ?
+        """,
+        (branch_db_id,),
+    )
```
[the remark this answers is not in the document]
Comment (NodeJSmith) [reply src/ccrecall/summary_enrichment.py:606]: Fixed - source-hash reads now filter for active branches.

File: src/ccrecall/transcript_sources.py [hunk 18]
```diff
@@ -0,0 +1,285 @@
...
+        visited.add(current)
+        with contextlib.suppress(OSError):
+            for child in sorted(current.iterdir()):
+                if child.name == "subagents":
+                    with contextlib.suppress(OSError):
+                        for _path in child.glob(f"*{session_uuid}*.jsonl"):
+                            return True
+                elif child.is_dir():
+                    pending.append(child)
+    return False
```
[the remark this answers is not in the document]
Comment (NodeJSmith) [reply src/ccrecall/transcript_sources.py:60]: Fixed - unsafe symlinked-project traversal no longer descends into symlink cycles.

File: src/ccrecall/transcript_sources.py [hunk 19]
```diff
@@ -0,0 +1,285 @@
...
+            else:
+                had_unsafe_path = True
+
+    unique: list[Path] = []
+    seen: set[Path] = set()
+    for path in found:
+        if path not in seen:
+            seen.add(path)
+            unique.append(path)
+    return SessionTranscriptDiscovery(files=unique, had_unsafe_path=had_unsafe_path)
```
[the remark this answers is not in the document]
Comment (NodeJSmith) [reply src/ccrecall/transcript_sources.py]: Fixed - consolidated order-preserving path deduplication behind one helper.

File: tests/test_backfill_llm_summaries.py [hunk 20]
```diff
@@ -0,0 +1,1781 @@
...
+    def test_worker_import_stays_on_llm_summary_db_boundary(self):
+        probe = """
+import sys
+import ccrecall.hooks.backfill_llm_summaries as worker
+assert worker.get_connection.__module__ == "ccrecall.llm_summary_db"
+assert "ccrecall.db" not in sys.modules
+"""
+        completed = subprocess.run([sys.executable, "-c", probe], capture_output=True, text=True)
+
+        assert completed.returncode == 0, completed.stderr
```
[the remark this answers is not in the document]
Comment (NodeJSmith) [reply tests/test_backfill_llm_summaries.py:210]: Fixed - the subprocess probe now has an explicit timeout and check behavior.

File: tests/test_backfill_llm_summaries.py [hunk 21]
```diff
@@ -0,0 +1,1781 @@
...
+        stale_marker.parent.mkdir(parents=True, exist_ok=True)
+        stale_marker.write_text("999999", encoding="utf-8")
+        monkeypatch.undo()
+
+        monkeypatch.setattr(worker, "load_settings", lambda: {"db_path": str(tmp_path / "unused.db")})
+        monkeypatch.setattr(worker, "setup_logging", lambda *_args, **_kwargs: logging.getLogger("test-worker"))
+        monkeypatch.setattr(worker, "_run", lambda **_kwargs: worker.EXIT_OK)
+
+        assert worker.run() == worker.EXIT_OK
+        assert not stale_marker.exists()
```
[the remark this answers is not in the document]
Comment (NodeJSmith) [reply tests/test_backfill_llm_summaries.py:1892]: Fixed - the stale PID test now redirects its runtime marker to tmp_path.

File: tests/test_db.py [hunk 22]
```diff
@@ -805,6 +860,54 @@ def test_migration_is_reentrant(self, tmp_path):
...
+            INSERT INTO branches (
+                session_id, leaf_uuid, is_active, context_summary, context_summary_json,
+                summary_version, embedding_version, embedding_model, summary_version_at_embed
+            ) VALUES (?, 'leaf-v6', 1, 'legacy summary', '{"topic":"legacy"}', 5, 2, 'embed-model', 5)
+            """,
+            (session_id,),
+        )
+        conn.execute("PRAGMA user_version = 6")
+        conn.commit()
+        conn.close()
```
[the remark this answers is not in the document]
Comment (NodeJSmith) [reply tests/test_db.py:962]: Fixed - the migration test now starts from a genuine v6 branches schema.

File: tests/test_db.py [hunk 23]
```diff
@@ -1576,6 +1691,28 @@ def test_memory_sync_does_not_import_heavy_deps(self):
...
+        """memory_setup.py must not import the LLM summarizer boundary."""
+        code = "import ccrecall.hooks.memory_setup\nimport sys\nassert 'ccrecall.llm_summarizer' not in sys.modules\n"
+        result = subprocess.run(
+            [sys.executable, "-c", code],
+            capture_output=True,
+            text=True,
+            timeout=30,
+        )
+        assert result.returncode == 0, result.stderr
+
```
[the remark this answers is not in the document]
Comment (NodeJSmith) [reply tests/test_db.py:1810]: Fixed - repeated import probes now share a helper with explicit subprocess behavior.

File: tests/test_llm_summarizer.py [hunk 24]
```diff
@@ -0,0 +1,1468 @@
...
+                claude_version="1.2.3",
+                run=ok_run,
+            )
+            sidecar_data = read_capability_sidecar(sidecar)
+
+            assert result.status == STATUS_OK
+            assert sidecar_data is not None
+            fingerprints.add(sidecar_data["fingerprint"])
+
+        assert len(fingerprints) == 1
```
[the remark this answers is not in the document]
Comment (NodeJSmith) [reply tests/test_llm_summarizer.py:448]: Covered - regressions now verify security-flag fingerprinting and nested unsafe state traversal.

File: tests/test_llm_summarizer.py [hunk 25]
```diff
@@ -0,0 +1,1468 @@
...
+
+        packet_parent = tmp_path / "packets"
+        stale = packet_parent / "stale"
+        live = packet_parent / "live"
+        stale.mkdir(parents=True)
+        live.mkdir(parents=True)
+        (stale / "manifest.json").write_text(
+            json.dumps({"pid": 999999, "created_at": "2026-01-01T00:00:00Z"}),
+            encoding="utf-8",
+        )
```
[the remark this answers is not in the document]
Comment (NodeJSmith) [reply tests/test_llm_summarizer.py:1603]: Fixed - the stale-packet test now controls PID liveness explicitly.

File: tests/test_llm_summary_evaluation.py [hunk 26]
```diff
@@ -0,0 +1,68 @@
...
+
+        assert len(session_ids) == 1
+        assert len(message_rows) >= 16, f"{scenario['id']} should stay long enough for review"
+        assert len(user_turns) >= 8, f"{scenario['id']} lost too many user turns"
+        assert len(assistant_turns) >= 8, f"{scenario['id']} lost too many assistant turns"
+        assert all(str(row.get("cwd", "")).startswith("/synthetic/") for row in message_rows)
+        serialized_fixture = fixture_path.read_text(encoding="utf-8")
+        assert "/home/" not in serialized_fixture
+        assert "/Users/" not in serialized_fixture
+        assert "C:\\Users\\" not in serialized_fixture
```
[the remark this answers is not in the document]
Comment (NodeJSmith) [reply tests/test_llm_summary_evaluation.py:61]: Fixed - fixture de-identification now rejects forward-slash Windows home paths.

File: tests/test_search.py [hunk 27]
```diff
@@ -1718,6 +1752,129 @@ def test_card_graceful_degrade_no_summary_json(self):
...
+            ),
+            (
+                json.dumps(_valid_search_enrichment()),
+                SUMMARY_ENRICHMENT_VERSION - 1,
+                "hash-current",
+                STATUS_OK,
+            ),
+        ],
+        ids=["invalid-envelope", "failed-status", "version-stale"],
+    )
```
[the remark this answers is not in the document]
Comment (NodeJSmith) [reply tests/test_search.py:1837]: Fixed - fallback coverage now includes a stale enrichment source hash.

File: tests/test_summary_enrichment.py [hunk 28]
```diff
@@ -0,0 +1,574 @@
...
+
+        assert rendered.startswith("### Branch Resume Brief")
+        assert "**Continuation hints:**" in rendered
+        assert "**How we got here:**" in rendered
+        assert "**Open questions:**" not in rendered
+        assert "### Session: deterministic" in rendered
+        assert sorted(ACTIVE_UUIDS)[0] not in rendered
+        llm_block, _, suffix = rendered.partition("\n\n" + base_markdown)
+        assert suffix == ""
+        assert len(llm_block) <= 2400
```
[the remark this answers is not in the document]
Comment (NodeJSmith) [reply tests/test_summary_enrichment.py]: Fixed - the render-budget assertion now uses PRIMARY_RENDER_BUDGET.

File: tests/test_sync_hook.py [hunk 29]
```diff
@@ -1022,3 +1022,218 @@ def raising_record(reason):
...
+        direct.write_text("{}\n", encoding="utf-8")
+        subagent = subagents / f"agent-{session_uuid}.jsonl"
+        subagent.write_text("{}\n", encoding="utf-8")
+        outside = tmp_path / "outside.jsonl"
+        outside.write_text("{}\n", encoding="utf-8")
+        (project_a / f"{session_uuid}-symlink.jsonl").symlink_to(outside)
+
+        found = sync_current.get_session_file(projects_dir, session_uuid)
+
+        assert found == direct
```
[the remark this answers is not in the document]
Comment (NodeJSmith) [reply tests/test_sync_hook.py:1197]: Fixed - the direct-transcript priority test is now independent of project path order.
